### PR TITLE
[Merged by Bors] - chore: add dates to more deprecations

### PR DESCRIPTION
**Pull request by grunweg**
And reformat nearby deprecations in the same file to use since syntax.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Init/Function.lean [hunk 1]
```diff
@@ -94,16 +94,16 @@ theorem id_comp (f : α → β) : id ∘ f = f := rfl
 #align function.left_id Function.id_comp
 #align function.comp.left_id Function.id_comp
 
-@[deprecated] alias left_id := id_comp -- Deprecated since 14 January 2014
-@[deprecated] alias comp.left_id := id_comp -- Deprecated since 14 January 2014
+@[deprecated] alias left_id := id_comp -- 2024-01-14
```
Comment (sgouezel) [inline Mathlib/Init/Function.lean:97]: Why not use the `since` syntax here and below?

File: Mathlib/Data/Complex/Exponential.lean [hunk 2]
```diff
@@ -210,6 +210,7 @@ noncomputable def expMonoidHom : MonoidHom (Multiplicative ℂ) ℂ :=
     map_one' := by simp,
     map_mul' := by simp [exp_add] }
 
+set_option linter.deprecated false in
```
Comment (sgouezel) [inline Mathlib/Data/Complex/Exponential.lean:213]: Why?

File: Mathlib/Init/Function.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Init/Function.lean]: Good idea, done.

**Comment by grunweg** [comment]
Thanks for the review; I have addressed all comments.
awaiting-review

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**